src/main/java/edu/unc/lib/boxc/migration/cdm/status/SourceFilesSummaryService.java

```diff
@@ -91,6 +92,96 @@ public int totalFilesInProject() {
         return getQueryService().countIndexedFileObjects();
     }
 
+    /**
+     * @return load list of all CSV records
+     */
+    public List<CSVRecord> loadAllEntries(Path mappingPath) {
```

Comment:
We briefly talked about trying to reduce the number of times the CSV file gets loaded, I think at the moment the new mappings file can end up being read 4-5 times per run which could slow things down for large migration projects. One way to address it would be:
1. Add a newStatePopulatedFiles instance variable
2. Introduce getter methods for newStatePopulatedFiles and previousStatePopulatedFiles which set/load their value using loadPopulatedEntries if it hasn't already been set
3. Use the getters in place of loadPopulatedEntries everywhere
4. For `countFilesMapped` we would probably need to change it to receiving a List<CSVRecord> instead of a Path, so that we could pass the correct list in instead of loading it again.

I think we'd be able to adjust `capturePreviousState` to just call `setPreviousStateFilesMapped` again since the loading of entries would be implicit when retrieving the previous state with the getter.